src/Riok.Mapperly/Emit/Syntax/SyntaxIndentationExtensions.cs

```diff
@@ -0,0 +1,94 @@
+        where TSyntax : SyntaxNode
+    {
+        using var enumerator = syntax.GetEnumerator();
+        if (!enumerator.MoveNext())
+            yield break;
+
+        yield return enumerator.Current!.AddLeadingLineFeed(indentation);
+
+        while (enumerator.MoveNext())
+        {
+            var node = enumerator.Current!;
+            yield return node.AddLeadingLineFeed(indentation).AddLeadingLineFeed(0);
+        }
+    }
+
+    /// <summary>
+    /// Adds a leading line feed to the first found trivia.
+    /// If the first token is known by the caller, use <see cref="AddLeadingLineFeed"/> for the first token instead
+    /// (should have better performance)
+    /// </summary>
+    /// <param name="syntax">The syntax.</param>
+    /// <param name="indentation">The indentation.</param>
+    /// <typeparam name="TSyntax">The type of the syntax.</typeparam>
+    /// <returns>The updated syntax.</returns>
+    public static TSyntax AddLeadingLineFeed<TSyntax>(this TSyntax syntax, int indentation)
+        where TSyntax : SyntaxNode
+    {
+        var trivia = syntax.GetLeadingTrivia();
+        trivia = trivia.AddLineFeedAndIndentation(indentation);
+        return syntax.WithLeadingTrivia(trivia);
+    }
+
+    /// <summary>
+    /// Adds a trailing line feed to the last found trivia.
+    /// If the last token is known by the caller, use <see cref="AddTrailingLineFeed"/> for the last token instead
+    /// (should have better performance).
+    /// </summary>
+    /// <param name="syntax">The syntax.</param>
+    /// <param name="indentation">The indentation.</param>
+    /// <typeparam name="TSyntax">The type of the syntax.</typeparam>
+    /// <returns>The updated syntax.</returns>
+    public static TSyntax AddTrailingLineFeed<TSyntax>(this TSyntax syntax, int indentation)
+        where TSyntax : SyntaxNode
+    {
+        var trivia = syntax.GetTrailingTrivia();
+        trivia = trivia.AddLineFeedAndIndentation(indentation);
+        return syntax.WithTrailingTrivia(trivia);
+    }
+
+    public static TSyntax AddTrailingSpace<TSyntax>(this TSyntax syntax)
+        where TSyntax : SyntaxNode => syntax.WithTrailingTrivia(syntax.GetTrailingTrivia().Add(ElasticSpace));
+
+    public static SyntaxToken AddLeadingLineFeed(this SyntaxToken token, int indentation)
+    {
+        var trivia = token.LeadingTrivia.AddLineFeedAndIndentation(indentation);
+        return token.WithLeadingTrivia(trivia);
+    }
+
+    public static SyntaxToken AddTrailingLineFeed(this SyntaxToken token, int indentation)
+    {
+        var trivia = token.TrailingTrivia.AddLineFeedAndIndentation(indentation);
+        return token.WithTrailingTrivia(trivia);
+    }
+
+    public static SyntaxTriviaList AddLineFeedAndIndentation(this SyntaxTriviaList trivia, int indentation)
+    {
+        trivia = trivia.Insert(0, ElasticCarriageReturnLineFeed);
+        for (var i = 0; i < indentation; i++)
```

Comment:
could `trivia.InsertRange` be user here? Roslyn is usually Immutable so this is likely allocating a lot of times. Plus inserting to the front of a list causes everything to copied as it is shifted.

Reply:
Added a cached trivia array / insert approach, slightly improved the benchmark.